plugins/module_utils/config_utils.py

```diff
@@ -99,3 +134,370 @@ def changed(self) -> bool:
+        - key (str): The key to search for in the dictionary.
+
+        Returns:
+        - The value associated with the found key, or None if the key is not found in the dict.
+
+        Example:
+        - Given the dictionary `{"interfaces": {"wan": {"if": 'interfaces/wan/if'}}}`
+        and the key `'if'`, the function will return `'interfaces/wan/if'`.
+        """
+
+        if isinstance(dictionary, dict):
+            for k, v in dictionary.items():
+                if k == key:
+                    return v
+                else:
+                    result = self._search_map(v, key)
+                    if result is not None:
+                        return result
+
+    def _get_xpath(self, module: str, setting: str) -> Union[str, dict]:
+        """
+        Retrieves the XPath for a given module and setting based on the version-specific mapping.
+
+        This function looks up the XPath in a version-specific mapping dictionary using the provided
+        module and setting names. If the setting is directly within the module, it returns it;
+        otherwise, it performs a recursive search if the setting is nested within the module.
+
+        Parameters:
+        - module (str): The name of the module for which the XPath is sought.
+        - setting (str): The specific setting within the module whose XPath is needed.
+
+        Returns:
+        - Union[str, dict, None]: The XPath as a string or dict if found, or None if the module
+        or setting cannot be found or is not available.
+
+        Raises:
+        - OPNSenseConfigUsageError: If the version-specific `version_map` is not set during the
+        initialization of the instance.
+
+        Example:
+        - Calling _get_xpath('network', 'interface') might return 'network/interface' or a
+        dictionary of nested settings if 'interface' is a nested setting within 'network'.
+
+        Note:
+        - The function emphasizes the importance of a properly initialized `version_map`.The absence
+        of the module in the map or the setting within the module leads to a return value of None.
+        """
+
+        if not self.version_map:
+            raise OPNSenseConfigUsageError(
+                "Module specific version_map was not set during initalization"
+            )
+
+        map_dict = self._get_module_version_map(module=module)
+
+        # Check if the provided module is in the map
+        if module in map_dict:
+            # If the setting is directly within the module, return it
+            if setting in map_dict[module]:
+                return map_dict[module][setting]
+            # If the setting is nested, search recursively
+            else:
+                return self._search_map(map_dict[module], setting)
+
+        raise OPNSenseConfigUsageError(
+            "Module specific xpath was not found"
+        )
+
+    def _get_php_requirements(self, module: str, setting: str) -> list:
```

Comment:
This looks like a special case of `_get_xpath`. Therefore the I don't think that the `settings` argument is required here. IMO it would suffice to just hard code `php_requirements` on line 285.  Also the setting is not expected to be nested therefore it could be simplified further.